File: movieshows/src/main/java/com/niranjandroid/movieshows/base/BaseContract.java
package com.niranjandroid.movieshows.base;

import android.support.v4.app.Fragment;

/**
 * Created by Niranjan P on 8/20/2017.
 */

public interface BaseContract {
    interface View {
        void init();

        void showToast(String message);

        void showNetworkError();
    }

    interface ActivityView extends View {
        void navigateToFragment(Fragment fragment, String tag);

        void navigateToFragment(Fragment fragment, String tag, boolean noHistory);

        void navigateToFragment(Fragment fragment, String tag, boolean noHistory, boolean clearStack);
    }

    interface FragmentView extends View {
        void launchFragment(Fragment fragment, String tag);
    }

    interface Presenter {
        void clear();

        void disposeComposites();
    }
}

File: movieshows/src/main/java/com/niranjandroid/movieshows/ui/base/BaseContract.kt
package com.niranjandroid.movieshows.ui.base

import android.support.v4.app.Fragment
import com.niranjandroid.movieshows.data.model.MovieListModel

/**
 * Created by Niranjan P on 8/20/2017.
 */

interface BaseContract {
    interface View {
        fun init()

        fun showToast(message: String)

        fun showNetworkError()
    }

    interface ActivityView : View {
        fun navigateToFragment(fragment: Fragment, tag: String)

        fun navigateToFragment(fragment: Fragment, tag: String, noHistory: Boolean)

        fun navigateToFragment(fragment: Fragment, tag: String, noHistory: Boolean, clearStack: Boolean)
    }

    interface FragmentView : View {
        fun launchFragment(fragment: Fragment, tag: String)
    }

    interface Presenter {
        fun clear()

        fun disposeComposites()
    }
}
